### PR TITLE
Bugfix -- set_optimizer is not implemented

### LINKED ISSUE #18514: set_optimizer does not exist in model.jl
model.jl line 446:
set_optimizer(kvstore, optimizer)

set_optimizer is not implemented, instead changing the line to:
setoptimizer!(kvstore, optimizer)

resolves the issue.

### PR DESCRIPTION
**Pull request by andevellicus**
Fixes #18514

## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by wkcn** [comment]
In Python front-end API, `kvstore.set_optimizer` is used.
`set_optimizer` and `setoptimizer`, which one is better?
I prefer the former, but the latter may be compatible with some MXNet applications in julia.

```bash
➜  julia git:(master) ✗ ag set_optimizer
NEWS.md
83:        * `set_optimizer()`

src/kvstore.jl
267:When running on multiple machines one must use `set_optimizer`.

src/model.jl
446:      set_optimizer(kvstore, optimizer)
```
```bash
➜  julia git:(master) ✗ ag setoptimizer
src/kvstore.jl
301:    setoptimizer!(kv::KVStore, opt)
321:julia> setoptimizer!(kv, SGD(η = .2))  # SGD with .2 as learning rate
338:function setoptimizer!(kv::KVStore, opt::AbstractOptimizer)

src/MXNet.jl
113:       setoptimizer!,
```

**Comment by leezu** [comment]
@andevellicus also note that MXNet 2 is deprecating the model API which Julia binding relies on here. You may want to target the 1.x branch with this PR?

**Comment by iblis17** [comment]
> In Python front-end API, kvstore.set_optimizer is used.
set_optimizer and setoptimizer, which one is better?
I prefer the former, but the latter may be compatible with some MXNet applications in julia.

yeah, the latter one matches the naming convention of Julia.

File: julia/src/model.jl [hunk 1]
```diff
@@ -443,7 +443,9 @@ function fit(self::FeedForward, optimizer::AbstractOptimizer, data::AbstractData
 
   if !isa(kvstore, Cvoid)
     if update_on_kvstore
-      set_optimizer(kvstore, optimizer)
+      #set_optimizer(kvstore, optimizer)
```
Comment (iblis17) [inline julia/src/model.jl:446]:
```suggestion
```

File: julia/src/model.jl [hunk 2]
```diff
@@ -443,7 +443,8 @@ function fit(self::FeedForward, optimizer::AbstractOptimizer, data::AbstractData
 
   if !isa(kvstore, Cvoid)
     if update_on_kvstore
-      set_optimizer(kvstore, optimizer)
+      setoptimizer!(kvstore, optimizer)
+
```
Comment (iblis17) [inline julia/src/model.jl:447]:
```suggestion
```

**Pull request closed without merging**